### PR TITLE
chore: switch to using subaccounts & deprecate tla creation

### PR DESCRIPTION
**Pull request by shariffdev**

### CODE REVIEW

**Comment by shariffdev** [comment]
@frol Ready for review

File: workspaces/tests/gas_meter.rs [hunk 1]
```diff
@@ -315,3 +228,61 @@ async fn test_dropped_gas_meter() -> anyhow::Result<()> {
...
+                }),
+        )
+        .unwrap();
+
+    Ok(account.into_result()?)
+}
+
+const fn as_near(gas: Gas) -> NearToken {
+    NearToken::from_yoctonear(gas.as_gas() as u128)
+}
```
Comment (frol) [inline workspaces/tests/gas_meter.rs:288]: What is going on here? You should not turn Gas into NearToken. Those are completely different things.

File: workspaces/tests/gas_meter.rs [hunk 2]
```diff
@@ -286,21 +197,23 @@ async fn test_gas_meter_batch_tx() -> anyhow::Result<()> {
...
-
-    // analogous to: worker.dev_create_account().await?;
-    let account = {
-        let (id, sk) = worker.dev_generate().await;
-        let account = worker.create_tla(id.clone(), sk).await?;
-        total_gas += account.details.total_gas_burnt.as_gas();
-
-        account.into_result()?
-    };
+    let mut total_gas = NearToken::from_yoctonear(0);
```
Comment (frol) [inline workspaces/tests/gas_meter.rs:200]: Gas has to be Gas, not NearToken

File: workspaces/src/types/mod.rs [hunk 3]
```diff
@@ -44,6 +44,9 @@ pub type BlockHeight = u64;
 /// Shard index, from 0 to NUM_SHARDS - 1.
 pub type ShardId = u64;
 
+// Constant taken from nearcore crate to avoid dependency
+pub const DEFAULT_DEPOSIT: NearToken = NearToken::from_near(100);
```
Comment (frol) [inline workspaces/src/types/mod.rs:48]: There is no sense in such a public constant. There is no such a thing like "default deposit" in NEAR protocol.

File: workspaces/src/network/variants.rs [hunk 4]
```diff
@@ -15,6 +15,7 @@ pub trait NetworkInfo {
     fn info(&self) -> &Info;
 }
 
+#[deprecated = "only the registrar can create top level accounts as of Protocol >=64"]
```
Comment (frol) [inline workspaces/src/network/variants.rs:18]: Top-level accounts creation is not deprecated, so I am not completely convinced that we should mark this trait as deprecated

**Review by frol — CHANGES_REQUESTED** [review]

File: workspaces/src/types/mod.rs (hunk 3 above)
Comment (shariffdev) [reply workspaces/src/types/mod.rs:48]: I moved this over to types crate: https://github.com/near/near-workspaces-rs/blob/51e0ead85556c57f3bd41398cd75025ad51c33b3/workspaces/src/network/sandbox.rs#L22

File: workspaces/src/network/variants.rs (hunk 4 above)
Comment (shariffdev) [reply workspaces/src/network/variants.rs:18]: I added that as an attempt to discourage usage, maybe we use `warn`?

File: workspaces/src/types/mod.rs (hunk 3 above)
Comment (shariffdev) [reply workspaces/src/types/mod.rs]: > There is no sense in such a public constant. There is no such a thing like "default deposit" in NEAR protocol.

Updated the docs and renamed the constant to reflect that.

**Comment by frol** [comment]
Resolved with #360

**Pull request closed without merging**